devops/providers/azure-devops/templates/azure-pipelines.yml

```diff
@@ -20,29 +22,64 @@ pr:
     include:
     - /infra/modules/*
     - /infra/templates/*
-    - /tests/*
+    - /test/*
     exclude:
     - devops/*
     - design-reference/*
-    - README.md
+    - /**/*.md
+    - /test/docker/*
 
 pool:
   vmImage: 'Ubuntu-16.04'
 
 variables:
-  GOBIN:  '$(GOPATH)/bin' # Go binaries path
-  GOROOT: '/usr/local/go1.11' # Go installation path
-  GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
-  modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code
+- group: "KV Secrets"
+- name: GO_VERSION
+  value: '1.11'
+- name: TF_VERSION
+  value: '0.11.13'
+- name: TEST_HARNESS_BASE_IMAGE_TAG
+  value: 'g$(GO_VERSION)t$(TF_VERSION)'
+- name: TF_STATE_STORAGE_ACCT
+  value: 'cobaltfstate'
+- name: TF_STATE_CONTAINER
+  value: 'cobaltfstate-remote-state-container'
+- name: TEST_HARNESS_BASE_IMAGE
+  value: 'msftcse/cobalt-test-base'
+- name: BUILD_TEST_RUN_IMAGE
+  value: 'cobalt-test-harness'
 
 steps:
+- checkout: self
+  persistCredentials: true
+  clean: true
+
+- script: |
+    docker login $ACR_HOST -u $ACR_USERNAME -p $(DOCKER-PASSWORD)
+
+- script: |
```

Comment:
So checked with AzDO there isnt a way to trigger tasks from repo path and this is a good approach to acheive this. Another approach would have been to have multiple build jobs based on repo file path  that would execute different parts of the test suite. Only concern here is it may get tricky to get the logs out running everything inside docker.